packages/core/src/declareAtom.ts

```diff
@@ -22,12 +22,15 @@ export const initAction = initActionCreator()
 type AtomName = TreeId | [string]
 type AtomsMap = { [key: string]: Atom<any> }
 type Reducer<TState, TValue> = (state: TState, value: TValue) => TState
-type DependencyMatcher<TState> = (
-  on: <T>(
+interface DependencyMatcherOn<TState> {
```

Comment:
Add a comment to the issue, pls (why we use an interface here)

Reply:
Ok, fixed